cmd/cloud/dbt.go

```diff
@@ -0,0 +1,154 @@
+	dbtBundleType             = "dbt"
+)
+
+func newDbtCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "dbt",
+		Short: "Manage your dbt projects deployed to Deployments running on Astronomer",
+	}
+	cmd.AddCommand(
+		newDbtDeployCmd(),
+	)
+	return cmd
+}
+
+func newDbtDeployCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "deploy DEPLOYMENT-ID",
+		Short:   "Deploy your dbt project to a Deployment on Astro",
+		Long:    "Deploy your dbt project to a Deployment on Astro. This command bundles your dbt project files and uploads it to your Deployment.",
+		Args:    cobra.MaximumNArgs(1),
+		RunE:    deployDbt,
+		Example: dbtDeployExample,
+	}
+
+	cmd.Flags().StringVarP(&mountPath, "mount-path", "m", "", "Path to mount dbt project in Airflow, for reference by DAGs. Default /usr/local/dbt/{dbt project name}")
+	cmd.Flags().StringVarP(&dbtProjectPath, "project-path", "b", "", "Path to the dbt project to deploy. Default current directory")
+	cmd.Flags().StringVar(&workspaceID, "workspace-id", "", "Workspace for your Deployment")
+	cmd.Flags().StringVarP(&deploymentName, "deployment-name", "n", "", "Name of the Deployment to deploy to")
+	cmd.Flags().StringVarP(&deployDescription, "description", "", "", "Description to store on the deploy")
+	cmd.Flags().BoolVarP(&waitForDeploy, "wait", "w", false, "Wait for the Deployment to become healthy before ending the command")
+
+	return cmd
+}
+
+func deployDbt(cmd *cobra.Command, args []string) error {
+	cmd.SilenceUsage = true
+
+	// if the dbt project path is not provided, use the current directory
+	if dbtProjectPath == "" {
+		dbtProjectPath = config.WorkingPath
+	}
+
+	// check that there is a valid dbt project at the dbt project path
+	dbtProjectYamlPath := filepath.Join(dbtProjectPath, dbtProjectYmlFilename)
+	if _, err := os.Stat(dbtProjectYamlPath); os.IsNotExist(err) {
+		return fmt.Errorf("dbt project not found in %s. Please run this command in the root of your dbt project, or use --project-path to specify the dbt project path.", dbtProjectYamlPath) //nolint
+	}
+	dbtProjectName, err := extractDbtProjectName(dbtProjectYamlPath)
+	if err != nil {
+		return fmt.Errorf("dbt project name not found in %s", dbtProjectYamlPath) //nolint
+	}
+
+	// if the workspace ID is not provided, try to find a valid workspace
+	if workspaceID == "" {
+		var err error
+		workspaceID, err = coalesceWorkspace()
+		if err != nil {
+			return errors.Wrap(err, "failed to find a valid workspace")
+		}
+	}
+
+	// get the deployment to deploy the dbt project to
+	var deploymentID string
+	if len(args) > 0 {
+		// if provided, use the deployment ID from the command argument
+		deploymentID = args[0]
+	} else {
+		// otherwise, prompt the user to select a deployment
+		selectedDeployment, err := deployment.GetDeployment(workspaceID, "", deploymentName, false, nil, platformCoreClient, astroCoreClient)
+		if err != nil {
+			return err
+		}
+		deploymentID = selectedDeployment.Id
+	}
```

Comment:
Is it worth extracting this into a common/shared function so that the two deploy commands always work the same?

Reply:
It's a fair question. In this change I really grappled with how much I wanted to unravel the existing deploy logic to fit in bundles. Frankly what is already there has over the years become a mess of poorly scoped function calls. This spot is a good example -- the existing deploy command has deployment resolution (including interactive prompting) not in the command code (where I would say it belongs) but buried deep in the service code in `deployment.GetDeployment`. Here this also calls that function so at least we're reusing that part.